src/main/scala/se/nimsa/dcm4che/streams/DicomPartFlow.scala

```diff
@@ -125,20 +125,20 @@ class DicomPartFlow(chunkSize: Int = 8192, stopTag: Option[Int] = None, inflate:
             case None =>
               InFmiHeader(updatedState)
           }
-          ParseResult(part, InValue(ValueState(updatedState.bigEndian, valueLength, nextStep)))
+          ParseResult(part, InValue(ValueState(updatedState.bigEndian, valueLength, nextStep)), acceptUpstreamFinish = false)
         }
       }
     }
 
     case class InDatasetHeader(state: DatasetHeaderState, inflater: Option[InflateData]) extends DicomParseStep {
       def parse(reader: ByteReader) = {
-        val attribute = readDatasetHeader(reader, state)
-        val nextState = attribute.map {
+        val part = readDatasetHeader(reader, state)
+        val nextState = part.map {
           case DicomHeader(_, _, length, _, bigEndian, _, _) => InValue(ValueState(bigEndian, length, InDatasetHeader(state, inflater)))
           case DicomFragments(_, _, bigEndian, _) => InFragments(FragmentsState(bigEndian, state.explicitVR), inflater)
           case _ => InDatasetHeader(state, inflater)
         }.getOrElse(FinishedParser)
-        ParseResult(attribute, nextState)
+        ParseResult(part, nextState, acceptUpstreamFinish = !nextState.isInstanceOf[InValue])
```

Comment:
This, an similar above fixes a bug where a `DicomHeader` at the end of a stream with value length 0 would not be followed by an empty `DicomValueChunk` as would be the case if this happened before the end.